script/generate-documentation

```diff
@@ -15,20 +15,47 @@
 # You should have received a copy of the GNU General Public License along
 # with this program; if not, see <http://www.gnu.org/licenses/>.
 
-# the script accepts input arguments:
+# The script runs in three major modes:
+# * only check that documentation generation doesn't show errors
+# * PUBLISH changes to upstream project
+# * PUBLISH changes to new branch in fork project and create Pull Request to
+# upstream project
+# 
+# First mode (PUBLISH is not set) is to run in CI as part of regular test
+# workflow
+# Second mode (PUBLISH is set) is to run manually from local environment when
+# need to publish documentation to upstream
+# Third mode (PULL_REQUEST_USER is set) is to run from CI nightly job
+
+# Example usages:
+# PUBLISH=1 script/generate-documentation https://github.com/andrii-suse/openQA.git gh-pages
+# PUBLISH=1 GITHUB_TOKEN=xxxx271e PULL_REQUEST_USER=andrii-suse script/generate-documentation https://token@github.com/openqabot/openQA.git gh-pages-$(date +%y%m%d%H%M%S)
+
+# The script accepts input arguments:
 # $1 - target repo (which repository is used for documentation commit), 
 # defaults to current repo
 # $2 - target branch, defaults to gh-pages
 
 # the script uses environment variables:
-# PUBLISH - non-empty string indicates that commit should be pushed back to target repo
-# (authentification must be set up in advance)
-# PULL_REQUEST_TARGET - if set, then will be passed to `hub pull-request` command
+# PUBLISH - non-empty string indicates that commit should be pushed back to
+# target repo (authentification must be set up in advance)
+# PULL_REQUEST_USER - if set, then will be appended with ':gh-pages' and
+# passed to  `hub pull-request` command
 # (authentification must be set up in advance)
+# If PULL_REQUEST_USER is set, then target branch ($2) must not exists at
+# target remote ($1). Target branch will be created based on
+# PULL_REQUEST_USER:gh_pages
 
 update_docs() {
-        git clone -b "$TARGET_BRANCH" --single-branch "$TARGET_REMOTE" out
+        mkdir out || exit 1
         cd out
+        if [[ -z "${PULL_REQUEST_USER}" ]]; then
+            hub clone --depth 1 --branch "$PUSH_BRANCH" "$PUSH_REMOTE_URL" .
```

Comment:
I like hub as well but should we rely on it in here as well? Do we need to? I don't see that you handle the dependency anywhere

Reply:
We must use hub for `pull-request` and it is easier to use it for `push`, so I decided that using it for `clone` will not hurt as well

Reply:
Assuming I'm reading it correctly it will hurt because even for a local build test you will need hub.

Reply:
I am fine with relying on hub but then we should ensure that we have it installed.

Reply:
I made sure that hub is required only for pull-request

Reply:
hm, sure, I can see that. But still, shouldn't we ensure it's installed? As we are relying on openSUSE as a base image how about a simple:

```
command -v hub >/dev/null || zypper -n in hub
```

before first use?

Reply:
It is installed in CI https://github.com/os-autoinst/openQA/pull/2440/files#diff-1d37e48f9ceff6d8030570cd36286a61R286 and I don't think that it may be acceptable to install such tools unconditionally in other environments.